src/sandstorm/control-socket.capnp

```diff
@@ -0,0 +1,75 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2020 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xd4c5feffbd79e908;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Util = import "util.capnp";
+
+interface Controller {
+  # Bootstrap interface offered by sandstorm on the control socket.
+  # This is used by some commands to control a running sandstorm
+  # server.
+  #
+  # This interface is not currently implemented. For now the commands
+  # referenced below use the `devmode` socket, and speak an ad-hoc
+  # protocol that provides similar functionality.
+  #
+  # The old protocol exists for historical reasons: when the devmode
+  # socket was first implemented, capnproto did not support passing
+  # file descriptors with a message. Now it does, so there is no
+  # need for a custom solution, and moving to capnproto will improve
+  # extensibility and maintainability.
+
+  const socketPath :Text = "/var/sandstorm/socket/control";
+  # Path to the control socket, relative to the sandstorm install (normaly
+  # /opt/sandstorm).
+
+  devShell @0 () -> (shellFds :List(FileDescriptor), cancel :Util.Handle);
+  # Stop the running sandstorm shell process, and get a list of file
+  # descriptors that the shell should inherit on startup.
+  #
+  # This is used by `sandstorm dev-shell`; the returned file descriptors
+  # will be passed to an instance of the shell launched from the developer's
+  # source tree, rather than from the installed sandstorm root.
+  #
+  # - `shellFds` is a list of file descriptors that the shell should inherit.
```

Comment:
When I read this my first question was: "What FD numbers should they be assigned? Like, does this include stdio FDs?" But looking at the code I see that's slightly complicated; it differs between dev mode and non-dev mode. Maybe worth a comment clarifying that these are sandstorm-specific FDs (not stdio) and the FD numbers to use depend on how you're invoking the shell.

Reply:
Hm, maybe it would be better to define a struct here that (roughly) corresponds to FdBundle, rather than just being an opaque list? The way I wrote this follows the current protocol closely, but the array feels like an implementation detail that on a second look I'm not sure makes sense anymore.

Reply:
Yeah that makes a lot of sense to me.

Reply:
Done.